### DISCUSSION TITLE
Live recording questions

### DISCUSSION THREAD

**Discussion by SirWaffle** [monocasual/giada#690] [General]
I'm new to giada, and liking it so far. I'm looking to add live sample recordings of guitar. I've gone over the documentation, and I'm wondering if I'm missing anything:

Is it possible to prevent overwriting a loop while recording? Such as arming a sample set to basic loop, and have it stop recording once it's the max size of the sequencers beats? Or continuously overwrite the sample instead of being additive?

Is it possible to have a sample set to start recording at the first beat, or a specified beat? 

Is there any way to set a sample to be a specific length, and stop recording after that length is met?

Thanks!

**Comment by gvnnz** [comment]
Hi @SirWaffle, thanks for your input!

> Is it possible to prevent overwriting a loop while recording? Such as arming a sample set to basic loop, and have it stop recording once it's the max size of the sequencers beats? Or continuously overwrite the sample instead of being additive?

Valid points here, none of them have been implemented so far. Specifically, the "have it stop recording once it's the max size of the sequencers beats" is called -  I guess - punch in/out. It's a great feature to add in future versions of Giada. For now you might want to try the "overdub protection" channel flag (in the channel's menu). If checked, the arm button gets disabled when the channel is not empty. 

> Is it possible to have a sample set to start recording at the first beat, or a specified beat?

You can start recording at the first beat with the "record on signal" mode. More info here, last paragraph: https://giadamusic.com/documentation-live-sampling

> Is there any way to set a sample to be a specific length, and stop recording after that length is met?

Kind of. With the "free loop-length" mode you start recording, then when you're done the sequencer will adjust to the length of your take. More details here: https://giadamusic.com/tutorial-002-free-loop-length-recording
